server/image/image.go

```diff
@@ -0,0 +1,78 @@
+package image
+
+import (
+	"context"
+	imagepkg "github.com/argoproj/argo-cd/v2/pkg/apiclient/image"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type server struct {
+	keychain authn.Keychain
+}
+
+func (s *server) GetImage(ctx context.Context, req *imagepkg.GetImageRequest) (*imagepkg.GetImageResponse, error) {
```

Comment:
What would you think are the security considerations associated with this request? Basically any authenticated user gets to make this request?

Reply:
I had considered this. I'd worked on the assumption you could inspect any image from the repo, but maybe we should only allow you to inspect images for apps you have access to.

Reply:
At my company, I think you can pull any image from our private repository. I'm not saying that's a good idea, just that engineers already have that ability.